server/request.go

```diff
@@ -244,7 +245,46 @@ func (r *RpcRequest) sendTxToRelay() {
 		r.logError("[sendTxToRelay] redis:SetTxSentToRelay failed: %v", err)
 	}
 
-	// remember that this tx based on from+nonce (for cancel-tx)
+	txTo := r.tx.To()
+	if txTo == nil {
+		r.writeRpcError("invalid target")
+		return
+	}
+
+	// Check if nonce is correct (spam protection)
+	_req := types.NewJsonRpcRequest(1, "eth_getTransactionCount", []interface{}{r.txFrom, "latest"})
+	_res, err := utils.SendRpcAndParseResponseTo(r.defaultProxyUrl, _req)
+	if err != nil {
+		r.logError("[sendTxToRelay] eth_getTransactionCount failed: %v", err)
+		r.writeHeaderStatus(http.StatusInternalServerError)
+		return
+	}
+	_nonceStr := ""
+	err = json.Unmarshal(_res.Result, &_nonceStr)
+	if err != nil {
+		r.logError("[sendTxToRelay] eth_getTransactionCount unmarshall failed: %v - result: %s", err, _res.Result)
+		r.writeHeaderStatus(http.StatusInternalServerError)
+		return
+	}
+	_nonceStr = strings.Replace(_nonceStr, "0x", "", 1)
+	_nonceBigInt := new(big.Int)
+	_nonceBigInt.SetString(_nonceStr, 16)
+	if r.tx.Nonce() != _nonceBigInt.Uint64() {
```

Comment:
This will fail txs when you have a pending tx waiting. Should we be storing nonces of pending outgoing txs and use the max of that/eth_getTransactionCount.

Reply:
good catch! yes we should :)

Reply:
Updated the code, pls review again :)